SimPEG/electromagnetics/static/utils/static_utils.py

```diff
@@ -552,132 +560,129 @@ def plot_3d_pseudosection(
+        ax = fig.add_axes([0.1, 0.1, 0.7, 0.8])
+        cax = fig.add_axes([0.85, 0.1, 0.03, 0.8])
 
-    # 3D scatter plot
-    if plane_points == None:
-
-        if vlim == None:
-            norm = mpl.colors.Normalize(vmin=dvec.min(), vmax=dvec.max())
-        else:
-            norm = mpl.colors.Normalize(vmin=vlim[0], vmax=vlim[1])
+    if vlim == None:
+        norm = mpl.colors.Normalize(vmin=dvec.min(), vmax=dvec.max())
+    else:
+        norm = mpl.colors.Normalize(vmin=vlim[0], vmax=vlim[1])
 
+    # Scatter plot
+    if plot_type == 'scatter':
         data_plot = ax.scatter(
-            midxy[:, 0],
-            midxy[:, 1],
-            midz,
-            dvec,
-            s=s,
+            locations[:, 0],
+            locations[:, -1],
+            marker_size,
             c=dvec,
-            depthshade=False,
             norm=norm,
-            **scatter_opts,
+            **scatter_opts
         )
-    else:
-        # Place in list if only one plane defined
-        if isinstance(plane_points[0], np.ndarray):
-            plane_points = [plane_points]
-
-        # Expand to list of only one plane distance for all planes
-        if isinstance(plane_distance, list) != True:
-            plane_distance = len(plane_points) * [plane_distance]
 
-        # Pre-allocate index for points on plane(s)
-        k = np.zeros(len(dvec), dtype=bool)
-        for ii in range(0, len(plane_points)):
-
-            p1, p2, p3 = plane_points[ii]
-            a, b, c, d = define_plane_from_points(p1, p2, p3)
-
-            k = k | (
-                np.abs(a * midxy[:, 0] + b * midxy[:, 1] + c * midz + d)
-                / np.sqrt(a ** 2 + b ** 2 + c ** 2)
-                < plane_distance[ii]
-            )
-
-        if np.all(k == 0):
-            raise Exception(
-                """No locations are within *plane_distance* of any plane(s)
-                defined by *plane_points*. Try increasing *plane_distance*."""
-            )
-
-        if vlim == None:
-            norm = mpl.colors.Normalize(vmin=dvec[k].min(), vmax=dvec[k].max())
-        else:
-            norm = mpl.colors.Normalize(vmin=vlim[0], vmax=vlim[1])
+    # Filled contour plot
+    elif plot_type == 'tricontourf':
 
-        data_plot = ax.scatter(
-            midxy[k, 0],
-            midxy[k, 1],
-            midz[k],
-            dvec[k],
-            s=s,
-            c=dvec[k],
-            depthshade=False,
+        # Make initial contour plot
+        data_plot = ax.tricontourf(
+            locations[:, 0],
+            locations[:, -1],
+            dvec,
             norm=norm,
-            **scatter_opts,
+            **tricontourf_opts,
         )
 
+    else:
+        raise NotImplementedError("plot_type must be 'scatter' or 'tricontourf'")
+        
+    # Use a filled polygon to mask everything above
+    # that has a pseudo-location above the positions
+    # for nearest electrode spacings    
+    
+    if mask_topography:
```

Comment:
How robust is this for guessing the topographic surface?

Reply:
It's pretty good. I tested it for peaks and valleys. I wanted to be sure for the latter case. And it's done based on the electrode locations and not topography so I don't believe there is sampling issue there.